src/causalmodel.jl

```diff
@@ -0,0 +1,73 @@
+using LightGraphs, NamedTupleTools
+
+import LightGraphs:
+    AbstractGraph, nv, ne,
+    edges, add_edge!, rem_edge!,
+    has_edge, inneighbors, outneighbors,
+    neighbors, all_neighbors,
+    indegree, outdegree
+
+import LightGraphs.SimpleGraphs: SimpleDiGraph, fadj, badj
+
+import Base.eltype
+
+export CausalModel, add_vertex, mechanism
+
+export nv, ne, edges, add_edge!, rem_edge!, 
+    has_edge, inneighbors, outneighbors,
+    neighbors, all_neighbors,
+    indegree, outdegree, eltype
+
+# :name -> variable_name and :func -> function or distribution
+const Variable = @NamedTuple{name::Symbol, func}
+
+"Causal model with `dag` representing the model and `scm` holds the name of the variables and the SCM"
+struct CausalModel{T} <: AbstractGraph{T}
+    dag::SimpleDiGraph{T}  # Representing the DAG of the causal model
+    scm::Dict{T,Variable}  # Name and function (or distribution) of each variable
+end
```

Comment:
I like this approach, this is even compatible with Soss' conditional models in the full `causalmodel.jl` file